accession-commons-monotonic-generator-jpa/src/main/java/uk/ac/ebi/ampt2d/commons/accession/generators/monotonic/MonotonicAccessionRecoveryAgent.java

```diff
@@ -0,0 +1,58 @@
+package uk.ac.ebi.ampt2d.commons.accession.generators.monotonic;
+
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.entities.ContiguousIdBlock;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.service.ContiguousIdBlockService;
+import uk.ac.ebi.ampt2d.commons.accession.persistence.jpa.monotonic.service.MonotonicDatabaseService;
+
+import java.time.LocalDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+public class MonotonicAccessionRecoveryAgent {
+    private final ContiguousIdBlockService blockService;
+    private final MonotonicDatabaseService monotonicDatabaseService;
+
+    public MonotonicAccessionRecoveryAgent(ContiguousIdBlockService blockService,
+                                           MonotonicDatabaseService monotonicDatabaseService) {
+        this.blockService = blockService;
+        this.monotonicDatabaseService = monotonicDatabaseService;
+    }
+
+    public void runRecovery(String categoryId, String applicationInstanceId, LocalDateTime lastUpdatedTime) {
```

Comment:
I feel like we should add logging statement or have a way to report how many blocks were processed and how many blocks were recovered in each category.

Reply:
added logger statements